### ISSUE TITLE
Take into account 2 types of address prefixes AU and AS

### ISSUE THREAD

**Issue by Ben-Rey** [massalabs/thyra#481]
**Context**
*Describe / explain why we should do this: motivations, context or other info. Keep it brief and simple - PM*
The core team implemented a new way of generating addresses. Some will start with AU (For a user) and the others with AS (for the smart-contracts) 

**User flow**
*Describe the user flow using user stories so the end result is super clear - PM*

**How to**
*List the step-by-step to get it do if needed - PM*
to be defined

**Technical details**
*Give the technical insights so anyone in the team can tackle the tasks - Dev*

**QA testing**
*Does this task require some QA tests ?*
*If yes, explain how to validate it*

**Comment by SlnPons** [comment]
@Ben-Rey this will be done with Testnet 20 at the end of the month, does it?

**Comment by Ben-Rey** [comment]
Yes, and the next testnet should be released beginning of next month

**Comment by SlnPons** [comment]
On Thyra's side, here are the changes / impacts: 
* on the transaction side
* on the wallet / wallet plugin side

Questions:
* does the length of the address is changing?

cc @Ben-Rey

**Comment by Ben-Rey** [comment]
> On Thyra's side, here are the changes / impacts: 
> * on the transaction side
> * on the wallet / wallet plugin side
> 
> Questions:
> * does the length of the address is changing?
> 
> cc @Ben-Rey 

Yes the length will be impacted, 1 more character

**Comment by SlnPons** [comment]
In that case, the workload due to this change is way different that if it was only a prefixe change. 
We'll list what needs to be changed in the code and see who can handle this on time.

**Comment by SlnPons** [comment]
@SebastienValla I can't test this issue. I have the following errors on my side: 


![Image](https://user-images.githubusercontent.com/109611779/223522647-d80ea714-f2f1-4bda-a118-efc5de9bf16d.png)

Here is what I did: 
1. git pull latest version of the main 
2. go run to launch thyra on testnet20  - no error on my console
3. on thyra wallet page, import a new wallet (using testnet staking wallet to get tokens in without asking) 
4. wallet create with new address on the front-end BUT:
* balance is 0 
* if I refresh to see the balance update, I have the errors on the screenshots above 

Could you tell me if I'm doing sth wrong? 
If this is a problem on the thyra wallet side, please come back to me without doing anything as we'll deprecate the code very soon. 
@Thykof @gregLibert fyi - it needed will do the job on the wallet plugin (which I can't install so I'm blocked to test this issue easily on the plugin side 😅 see https://github.com/massalabs/thyra-plugin-wallet/issues/38)

**Comment by SlnPons** [comment]
Thanks @SebastienValla to unlock me. The problem was that Ive got wallets with old addresses --> when you delete them from the front-end it doesn't delete them from the config folder of your computer.... Will create an issue in the plugin wallet to solve this problem. 
Still blocked to test as no DNS is deployed on Testnet20

**Comment by Thykof** [comment]
> when you delete them from the front-end it doesn't delete them from the config folder of your computer

Maybe we don't need to write code for retro compatibility during testnet?
